src/components/Library.vue

```diff
@@ -175,52 +180,85 @@ export default {
     * CREATE OBJECT
     * initializes an empty object
     */
-    createObject() {
-      switch (this.mode) {
-        case 'stories':
-          this.$store.dispatch('models/initStory');
-          return;
-        case 'spaces':
-          this.$store.dispatch('models/initSpace', { story: this.currentStory });
-          break;
-        case 'shading':
-          this.$store.dispatch('models/initShading', { story: this.currentStory });
-          break;
-        case 'images':
-          window.eventBus.$emit('uploadImage');
-          break;
-        case 'windows':
-        case 'daylighting_controls':
-          window.eventBus.$emit('error', 'Create components by clicking where you would like it to be');
-          break;
-        default:
-          this.$store.dispatch('models/createObjectWithType', { type: this.mode });
-          break;
+    createObject({ duplicate }) {
+      if (duplicate) {
+        const height = this.currentStory.floor_to_ceiling_height;
+        this.$store.dispatch('models/initStory');
+        this.$store.dispatch('models/updateStoryWithData', { story: this.currentStory, floor_to_ceiling_height: height });
+        this.$store.dispatch('application/setCurrentTool', { tool: 'Rectangle' });
+        return;
+      } else {
+        switch (this.mode) {
+          case 'stories':
+            this.$store.dispatch('models/initStory');
+            return;
+          case 'spaces':
+            this.$store.dispatch('models/initSpace', { story: this.currentStory });
+            break;
+          case 'shading':
+            this.$store.dispatch('models/initShading', { story: this.currentStory });
+            break;
+          case 'images':
+            window.eventBus.$emit('uploadImage');
+            break;
+          case 'windows':
+          case 'daylighting_controls':
+            window.eventBus.$emit('error', 'Create components by clicking where you would like it to be');
+            break;
+          default:
+            this.$store.dispatch('models/createObjectWithType', { type: this.mode });
+            break;
+        }
       }
       this.selectLatest();
     },
     async duplicateRow(row) {
-      this.createObject();
-      await this.$nextTick();
-      let newName = row.name;
-      // increment Copy # until no name conflict.
-      while (_.find(this.rows, { name: newName })) {
-        let copyNum = newName.match(/\((\d+)\)/);
-        newName = copyNum ? `${newName.slice(0, copyNum.index)}(${+copyNum[1] + 1})` : `${newName} (2)`;
+      if(this.selectedObject.name.includes('Story')) {
```

Comment:
A user could rename a story so that it doesn't include 'Story'

Reply:
Good call - thank you.

Reply:
I'm not sure what to key this off of.

Reply:
Used the id of the selectedObject and check to see if I can find it in the list of stories.